### PR TITLE
Enhancement: Add Year to Job Run Timestamp in pipeline-stage-view-plugin

### PR DESCRIPTION
**Pull request by DeepData-Diver**
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by DeepData-Diver** [comment]
Addressed this Jenkins issue: https://issues.jenkins.io/browse/JENKINS-73507

File: pom.xml [hunk 1]
```diff
@@ -59,7 +59,7 @@
         <revision>2.35</revision>
         <gitHubRepo>jenkinsci/pipeline-stage-view-plugin</gitHubRepo>
         <bom>2.361.x</bom>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.462.1</jenkins.version>
```
Comment (jglick) [inline pom.xml:62]: Mismatch. See https://github.com/jenkinsci/archetypes/pull/737. Anyway this does not seem related to the rest of the PR and would better be reverted (there could be a separate PR to refresh dependencies).

File: pom.xml.versionsBackup
Comment (jglick) [inline pom.xml.versionsBackup:1]: Delete.

File: ui/src/main/js/view/templates/index.js [hunk 2]
```diff
@@ -100,6 +100,10 @@ registerHBSHelper('formatDate', function (date, toFormat) {
     if (toFormat == 'time') {
         return theDate.toLocaleTimeString(userLocale, {hour: '2-digit',minute: '2-digit', hour12: false });
     }
+    // New case for 'year'
```
Comment (jglick) [inline ui/src/main/js/view/templates/index.js:103]:
```suggestion
```

**Review by jglick — CHANGES_REQUESTED** [review]
Please be aware that this plugin is generally not maintained.

File: pom.xml.versionsBackup
Comment (DeepData-Diver) [reply pom.xml.versionsBackup:1]: Will delete tomorrow. Thank you

File: pom.xml (hunk 1 above)
Comment (DeepData-Diver) [reply pom.xml:62]: Will revert tomorrow. Thank you

Comment (DeepData-Diver) [reply pom.xml:62]: Done!

File: pom.xml.versionsBackup
Comment (DeepData-Diver) [reply pom.xml.versionsBackup:1]: Done!

File: pom.xml (hunk 1 above)
Comment (DeepData-Diver) [reply pom.xml]: Done in the sense that the changes have been reverted back. Thanks

**Comment by jglick** [comment]
I am not sure this is a net benefit; it breaks the layout a bit.

Chrome before

Screenshot from 2024-09-16 12-22-16

and after

Screenshot from 2024-09-16 12-23-01

and Firefox before

Screenshot 2024-09-16 at 12-27-05 x Jenkins

and after

Screenshot 2024-09-16 at 12-27-13 x Jenkins

**Comment by jglick** [comment]
Note that the issue suggests showing the year only if the build is more than a year old, which seems prudent. (Or a few months old at least. For that matter, maybe we should display the _day_ only if the build is more than a few hours old. Otherwise the time would suffice.)

**Comment by DeepData-Diver** [comment]
Addressed PR comments. Please see new changes. Here is a sample image that illustrates changes:
image

**Comment by DeepData-Diver** [comment]
@jglick Please review PR. Thanks

**Comment by DeepData-Diver** [comment]
Hello, just checking to see if more changes are required for this PR?

**Comment by jglick** [comment]
Did not check; nobody maintains this plugin.

**Comment by olamy** [comment]
let's get this merged

**Comment by fahol-coop** [comment]
The last build broke the UI, as already mentioned before. The date field is now empty (blank space)
image

**Comment by lppedd** [comment]
Same here. I'll revert to 2.34 in the meantime.

image

**Comment by lppedd** [comment]
Downgrading is trickier than what I expected. I can downgrade the Stage plugin, but not the REST one.

**Comment by lppedd** [comment]
Folks, can we revert this? I can quickly fork and open a PR. This has caused more downsides than upsides.

**Comment by olamy** [comment]
@lppedd sure please do it. Thanks

**Comment by maichouni-mitek** [comment]
> Folks, can we revert this? I can quickly fork and open a PR. This has caused more downsides than upsides.

@lppedd , have you managed to downgrade? or is the PR still the route to go?
Thanks.
